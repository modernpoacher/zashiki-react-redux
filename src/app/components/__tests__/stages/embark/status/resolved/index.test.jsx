
import React from 'react'
import renderer from 'react-test-renderer'

import Resolved from '@modernpoacher/zashiki-react-redux/app/components/stages/embark/status/resolved'

const MOCK_DEFINITION = {
  meta: {
    uri: '#/'
  },
  elements: {
    title: 'MOCK TITLE',
    field: {
      id: 'MOCK ID'
    }
  }
}

const MOCK_RESPONSE = {
  '#/': 'MOCK VALUE'
}

describe('@modernpoacher/zashiki-react-redux/app/components/stages/embark/status/resolved', () => {
  describe('Always', () => {
    it('renders', () => {
      const component = (
        <Resolved
          definition={MOCK_DEFINITION}
          response={MOCK_RESPONSE}
          errors={[]}
          onChange={jest.fn()}
          onSubmit={jest.fn()}
        />
      )

      expect(renderer.create(component).toJSON())
        .toMatchSnapshot()
    })
  })
})
