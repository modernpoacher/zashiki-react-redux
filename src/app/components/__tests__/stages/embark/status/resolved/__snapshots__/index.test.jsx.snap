// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`#zashiki-react-redux/app/components/stages/embark/status/resolved Always renders 1`] = `
<div
  className="embark resolved"
>
  <Stage>
    <form
      action="/embark-stage"
      method="POST"
      onSubmit={[Function]}
    >
      <ErrorSummarySprocket />
      <Pinion>
        <Field>
          <TextCog>
            <div
              className="cog text"
            >
              <TextTitle>
                <label
                  className="title text"
                  htmlFor="MOCK ID"
                >
                  <TextContent>
                    <span
                      className="text-content"
                    >
                      MOCK TITLE
                    </span>
                  </TextContent>
                </label>
              </TextTitle>
              <TextDescription>
                <span
                  className="description text"
                >
                  <TextContent>
                    <span
                      className="text-content"
                    >
                      MOCK DESCRIPTION
                    </span>
                  </TextContent>
                </span>
              </TextDescription>
              <TextErrorMessage />
              <TextField>
                <input
                  className="input text"
                  disabled={false}
                  id="MOCK ID"
                  name="MOCK ID"
                  onChange={[Function]}
                  readOnly={false}
                  required={false}
                  type="text"
                  value=""
                />
              </TextField>
            </div>
          </TextCog>
        </Field>
      </Pinion>
      <button
        type="submit"
      >
        Continue
      </button>
    </form>
  </Stage>
</div>
`;
